### ISSUE TITLE
When creating instance placeholder programaticlly values don't get stored correctly

### ISSUE THREAD

**Issue by afk-mario** [godotengine/godot#73834]
### Godot version

v3.5.1.stable.arch_linux

### System information

Arch Linux x86_64 5.15.37-1-lts

### Issue description

I'm trying to set a packed scene as an Instance placeholder using a tool script, but calling:

`packedscene.set_scene_instance_load_placeholder(true)`

 keeps the original packed scene script crashing the project.

If I clear the script using:

`packedscene.script = null`

will loose all the non default values from the packed scene

### Steps to reproduce

Generate a node tree using a tool script and then try to convert a packed scene to a Instance Placeholder
```gdscript
tool
extends Node2D

export var create_enemies = false setget do_create_enemies

var enemy_scene := preload("res://enemy.tscn")


func _ready() -> void:
	if not Engine.editor_hint:
		for child in $wrapper.get_children():
			print(child.name)
			if child is InstancePlaceholder:
				print(child.name)
				print(child.get_stored_values())
				child.create_instance()

func do_create_enemies(_value) -> void:
	print("Creating enemies")
	var enemy = enemy_scene.instance()
	enemy.damage = 10
	enemy.position = Vector2(50,50)
	enemy.name = "enemy"
	enemy.set_scene_instance_load_placeholder(true)
	# Crashes without clearing the script
	enemy.script = null
	# but now the stored values are not saved correctly
	$wrapper.add_child(enemy)
	enemy.set_owner(self)
	print(enemy)

```

### Minimal reproduction project

[instance-placeholder.zip](https://github.com/godotengine/godot/files/10817970/instance-placeholder.zip)

**Comment by KoBeWi** [comment]
No errors in 4.0.

**Comment by tom-jk** [comment]
Having the same issue in 4.1.2, can also reproduce in master.

To test, you should first remove the ```enemy.script = null``` line. Then select main node and click Create Enemies in the inspector. Save the scene. Open it in a text editor and you should see:

```
[node name="enemy" type="Node2D" parent="wrapper" instance_placeholder="res://enemy.tscn"]
position = Vector2(50, 50)
script = ExtResource("2_pi0ae")
damage = 10
```

Run the scene. Godot will immediately stop with the error `Script inherits from native type 'Node2D', so it can't be assigned to an object of type: 'InstancePlaceholder'`.

Interestingly, if you now go `Scene` -> `Reload Saved Scene`, then save the scene again, in the text editor you'll see:

```
[node name="enemy" parent="wrapper" instance_placeholder="res://enemy.tscn"]
position = Vector2(50, 50)
damage = 10
```

The scene runs correctly now. The reload-save cycle flushed out the broken stuff, and now it's the same as what you get when you set placeholder from the context menu.

As afk-mario said, if you set script to null, the error goes away, but so do the properties associated with the script:

```
[node name="enemy" type="Node2D" parent="wrapper" instance_placeholder="res://enemy.tscn"]
position = Vector2(50, 50)
```

(It also still has `type="Node2D"`, which disappears with a reload-save cycle.)

----

In summary, either `Node.set_scene_instance_load_placeholder` or its documentation is incomplete.

**Comment by Shou** [comment]
This issue also arises when using `ResourceSaver` to create scenes that have `InstancePlaceholder` child nodes. At least in 3.6, but it might remain an issue in 4.x.

The code in question is a tool script that duplicates nodes (A) from the current scene tree, assigns node (A) to all of its children's `owner`, and then saves said node (A) using `ResourceSaver`. Exact same symptom of having an extra `script` field in the tres resource file causing the same "can't be assigned to an object of type: 'InstancePlaceholder'` error.

A hacky workaround that can work for both cases is to read the relevant scene resource file (not enemy.tscn, but the root node of the tool script) as text, line-by-line, and remove all `script = ExtResource` lines that follow a `[node ... instance_placeholder="..."]` by omitting them when you rewrite the scene tres. You'll likely want to do this from another tool-script scene, I don't think a scene rewriting itself is a good idea.
